plugins/module_utils/network/common/network.py

```diff
@@ -236,7 +236,7 @@ def get_resource_connection(module):
 
     capabilities = get_capabilities(module)
     network_api = capabilities.get("network_api")
-    if network_api in ("cliconf", "nxapi", "eapi", "exosapi"):
+    if network_api in ("cliconf", "nxapi", "eapi", "exosapi", "sonic_rest"):
```

Comment:
I think we should default to using `Connection` class. Adding an entry here for each transport variant is not scalable.
```
    if network_api == "netconf":
        module._connection = NetconfConnection(module._socket_path)
    elif network_api == "local":
        # This isn't supported, but we shouldn't fail here.
        # Set the connection to a fake connection so it fails sensibly.
        module._connection = LocalResourceConnection(module)
    else:
        module._connection = Connection(module._socket_path)
```

Reply:
updated as suggested